src/Features/VisualBasic/Portable/BraceCompletion/InterpolationBraceCompletionService.vb

```diff
@@ -0,0 +1,89 @@
+
+<Export(LanguageNames.VisualBasic, GetType(IBraceCompletionService)), [Shared]>
+Friend Class InterpolationBraceCompletionService
+    Inherits AbstractBraceCompletionService
+
+    <ImportingConstructor>
+    <Obsolete(MefConstruction.ImportingConstructorMessage, True)>
+    Public Sub New()
+        MyBase.New()
+    End Sub
+
+    Protected Overrides ReadOnly Property OpeningBrace As Char
+        Get
+            Return CurlyBrace.OpenCharacter
+        End Get
+    End Property
+
+    Protected Overrides ReadOnly Property ClosingBrace As Char
+        Get
+            Return CurlyBrace.CloseCharacter
+        End Get
+    End Property
+
+    Protected Overrides Function IsValidOpenBraceTokenAtPositionAsync(token As SyntaxToken, position As Integer, document As Document, cancellationToken As CancellationToken) As Task(Of Boolean)
+        Return Task.FromResult(IsValidOpeningBraceToken(token))
+    End Function
+
+    Public Overrides Function AllowOverTypeAsync(context As BraceCompletionContext, cancellationToken As CancellationToken) As Task(Of Boolean)
+        Return CheckClosingTokenKindAsync(context.Document, context.ClosingPoint, cancellationToken)
+    End Function
+
+    Public Overrides Async Function IsValidForBraceCompletionAsync(brace As Char, openingPosition As Integer, document As Document, cancellationToken As CancellationToken) As Task(Of Boolean)
+        Return OpeningBrace = brace And Await IsPositionInInterpolationContextAsync(document, openingPosition, cancellationToken).ConfigureAwait(False)
+    End Function
+
+    Protected Overrides Function IsValidOpeningBraceToken(token As SyntaxToken) As Boolean
+        Return token.IsKind(SyntaxKind.DollarSignDoubleQuoteToken, SyntaxKind.InterpolatedStringTextToken) OrElse
+                   (token.IsKind(SyntaxKind.CloseBraceToken) AndAlso token.Parent.IsKind(SyntaxKind.Interpolation))
+    End Function
+
+    Protected Overrides Function IsValidClosingBraceToken(token As SyntaxToken) As Boolean
+        Return token.IsKind(SyntaxKind.CloseBraceToken)
+    End Function
+
+    Public Shared Async Function IsPositionInInterpolationContextAsync(document As Document, position As Integer, cancellationToken As CancellationToken) As Task(Of Boolean)
+        If position = 0 Then
+            Return False
+        End If
+
+        ' Check to see if the character to the left of the position is an open curly brace. Note that we have to
+        ' count braces to ensure that the character isn't actually an escaped brace.
+        Dim text = Await document.GetTextAsync(cancellationToken).ConfigureAwait(False)
+        Dim index = position - 1
+        Dim openCurlyCount = 0
+        For index = index To 0 Step -1
+            If text(index) = "{"c Then
+                openCurlyCount += 1
+            Else
+                Exit For
+            End If
+        Next
+
+        If openCurlyCount Mod 2 > 0 Then
+            Return False
+        End If
```

Comment:
=-o.  wat

Reply:
this is similar to the logic in the c# version... unifying